assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -0,0 +1,186 @@
+import { Component, Fragment } from '@wordpress/element';
+import { ArrowIcon } from './ArrowIcon';
+
+const {apiFetch} = wp;
+const {addQueryArgs} = wp.url;
+const {__} = wp.i18n;
+
+export class SpreadsheetFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      sheet_data: null,
+      invalidSheetId: false,
+      searchText: '',
+      sortDirection: 'asc',
+      sortColumnIndex: 0,
+    };
+
+    this.onHeaderClick = this.onHeaderClick.bind(this);
+  }
+
+  onHeaderClick( index ) {
+    if ( index == this.state.sortColumnIndex ) {
```

Comment:
Hm I think so, I'll remove it and test.

Reply:
So it turns out this is actually needed to set the sortDirection to `asc` in case you clicked in a different header, otherwise it could be `desc` and trigger the animation on click. Anyway, Magali and I will review the header behavior as the next iteration, so this will probably need some changes.